src/main/java/online/book/store/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,14 @@
+package online.book.store.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import online.book.store.dto.UserRegistrationRequestDto;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+
+    @Override
+    public boolean isValid(Object object, ConstraintValidatorContext constraintValidatorContext) {
+        UserRegistrationRequestDto requestDto = (UserRegistrationRequestDto) object;
```

Comment:
Let's specify FieldMatchValidator with UserRegistrationRequestDto type and avoid casting